TestFoundation/TestIOStream.swift

```diff
@@ -0,0 +1,462 @@
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+
+#if DEPLOYMENT_RUNTIME_OBJC || os(Linux)
+    import Foundation
+    import XCTest
+#else
+    import SwiftFoundation
+    import SwiftXCTest
+#endif
+
+class TestIOStream : XCTestCase {
```

Comment:
This is tests for "has-a" value type wrapper a lot of the tests overlaps the "is-a" implementation
suggestions on weather they are needed are welcome.

any general suggestions about wether should we test at the wrapper level or at the reference level in general are welcomed here too